tests/cpp/test_multidevice_lower_communication.cpp

```diff
@@ -186,9 +186,26 @@ INSTANTIATE_TEST_SUITE_P(
          {{1, 2}, {0, 1}}, //
          {{1, 2}, {1, 0}}})));
 
-using LowerCollectiveTest = MultiDeviceTest;
+class LowerCollectiveTest
+    : public MultiDeviceTest,
+      public testing::WithParamInterface<CommunicatorBackend> {
+ protected:
+  void SetUp() override;
+};
+
+void LowerCollectiveTest::SetUp() {
+  MultiDeviceTest::SetUp();
+
+  const CommunicatorBackend backend_type = GetParam();
+  if (!communicator_->isBackendAvailable(backend_type)) {
+    GTEST_SKIP() << "Backend not available: " << backend_type;
+  }
+  // getBackendForTeam throws an error if the requested backend type isn't
+  // available. Therefore, we call it after the isBackendAvailable check.
+  communicator_->setDefaultBackend(backend_type);
```

Comment:
I discovered that while this allows me to set the backend when using FusionExecutorCache, `communication->backend()` is still different from the `backend` parameter passed to `postSingleCommunciation`.
`getBackendForTeam` [when not provided a backend returns the default backend set for the communicator](https://github.com/NVIDIA/Fuser/blob/53f0aceed858ed8d5842de1136303c139a94326c/csrc/host_ir/executor.cpp#L127-L128).
For executions running through `HostIrEvaluator`, this approach will not work.

We should change this to have a uniform value of backend between communicator and communication object for easier verification. I'll attempt this in a future PR.